tianshou/data/batch.py

```diff
@@ -16,10 +16,10 @@
 
 def _is_batch_set(data: Any) -> bool:
     if isinstance(data, (list, tuple)):
-        if len(data) > 0 and isinstance(data[0], (dict, Batch)):
+        if len(data) > 0 and all(isinstance(e, (dict, Batch)) for e in data):
```

Comment:
It seems heavily time consuming. Please post some profiling to check the performance.

Reply:
``isinstance `` is a quick check. The most time is spent on data copy, environment simulation, and learning.

Reply:
Not really. If you have a look to the profiling graph, we are already spending about 7% of the whole execution time checking types.

Reply:
Well, we have to recognize that current tests are some toy examples. In reality, the reinforcement learning task can run for days. In that practical usage, these type checks are almost free. But without these type check, a program written to run for days can break during training, resulting in a waste of several days.

Reply:
I don't fully agree. Even though this analysis is based on toy models, I don't think it is that far from the reality. And if it is, we should implement a proper benchmark for profiling Tianshou. Next, it is extremely likely to fail way before several days. Finally, reinforcement learning tasks running for days are not real. It does not exist outside the dream world outside universities and the GAFAM. Usually, companies using reinforcement learning are consuming much less resources in average that you seem imagine.

Reply:
I think maybe we can change the code to avoid too many calls on _is_batch_set.

Reply:
I guess we solved this issue by reducing the number of calls to this method.

Reply:
By the way, I found that the unittest total time goes back to less than 6 mins.